test/integration/framework.go

```diff
@@ -107,23 +109,27 @@ func getFreshApiserverAndClient(t *testing.T, storageTypeStr string) (servicecat
 	return clientset, shutdown
 }
 
-func waitForApiserverUp(insecureAddr string, stopCh <-chan struct{}) error {
-	minuteTimeout := time.After(2 * time.Minute)
-	for {
-		select {
-		case <-stopCh:
-			return fmt.Errorf("apiserver failed")
-		case <-minuteTimeout:
-			return fmt.Errorf("waiting for apiserver timed out")
-		default:
-			glog.Infof("Waiting for : %#v", insecureAddr)
-			_, err := http.Get(insecureAddr)
-			if err == nil {
-				return nil
+func waitForApiserverUp(serverURL string, stopCh <-chan struct{}) error {
+	startWaiting := time.Now()
+	tries := 0
+	// roughly 30 seconds total
+	return wait.ExponentialBackoff(wait.Backoff{1 * time.Second, 2, 0, 5},
```

Comment:
can you `PollImmediate` here? It makes more sense for a test that we proceed as soon as possible, at the expense of some CPU and allocations